tools/Makefile

```diff
@@ -28,4 +32,12 @@ tools-clean:
 	$(RM) tools/headerversions
 	$(RM) tools/headerversions.o
 
+tools/pip-install:
+	@if which $(PIP_EXE) >> /dev/null; then \
+		echo "installing python dependencies"; \
+		$(PIP_EXE) $(PIP_ARGS) install -r tools/requirements.txt ; \
+	else \
+		(echo "Pip not installed, skipping python dependencies") ; \
+	fi
```

Comment:
Ok, I see where this comes from, but it most likely isn't what we want:

 - Developer dependencies are tracked in `pyproject.toml`
 - There is currently no user dependencies (other than libraries from the distro), so we need something here. On the other hand the users will not be the ones to run `Makefile`s, so we better look for some better option. Alternatively we could just use `urllib3` and not have dependencies at all.

Potentially reckless could be a completely separate tool that we distribute on PyPI where it can have its own dependencies, and users would just be a `pip install reckless` away from being able to run it.

Reply:
I'm certainly open to a better option for installing python dependencies for the user (or at least prompting a `pip install`.)  I think we'll want that at some point.  Let's just switch to `urllib3` for the time being?  Distributing reckless via PyPI is fine, but I would still prefer it to be updated and working by default with the latest CLN release if possible.